plugins/processors/noise/noise_test.go

```diff
@@ -0,0 +1,165 @@
+package noise
+
+import (
+	"math"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/rand"
+	"gonum.org/v1/gonum/stat/distuv"
+)
+
+// Helper function which returns a map for metrics for easy lookup
+func createTestMetrics() map[string]telegraf.Metric {
+	m := make(map[string]telegraf.Metric)
+	m["cpu"] = testutil.MustMetric(
+		"cpu",
+		map[string]string{},
+		map[string]interface{}{
+			"usage_guest":  2.5,
+			"usage_system": 1.5,
+			"usage_nice":   0.5,
+			"usage_irq":    1.0,
+		},
+		time.Unix(0, 0),
+	)
+	m["disk"] = testutil.MustMetric(
+		"disk",
+		map[string]string{},
+		map[string]interface{}{
+			"free":        250,
+			"inodes_free": 1500.0,
+			"inodes_used": 1337,
+		},
+		time.Unix(0, 0),
+	)
+	return m
+}
+
+// Verifies that field values are modified by the Laplace noise
+func TestAddNoiseToMetric(t *testing.T) {
+	generators := []string{"laplace", "gaussian", "uniform"}
+	for _, generator := range generators {
+		p := Noise{
+			NoiseType: generator,
+			Scale:     1.0,
+			Mu:        0.0,
+			Sigma:     0.2,
+			Min:       -1,
+			Max:       1,
+			Log:       testutil.Logger{},
+		}
+		_ = p.Init()
+		metrics := testutil.MockMetrics()
+		after := p.Apply(metrics[0].Copy())[0]
+		require.NotEqual(t, metrics[0], after)
+	}
+}
+
+// Verifies that no int64 or uint64 overflow occurs while adding noise
+func TestAddNoiseOverflowCheck(t *testing.T) {
+	p := Noise{
+		NoiseType: "laplace",
+		Scale:     1.0,
+		Log:       testutil.Logger{},
+	}
+	_ = p.Init()
+	p.Generator = &distuv.Laplace{Mu: p.Mu, Scale: p.Scale, Src: rand.NewSource(4)}
+	require.Equal(t, math.MaxInt64, p.addNoise(math.MaxInt64))
+
+	// rand should be -1.3528540286500519, a underflow happens for uint64
+	require.Equal(t, uint64(0), p.addNoise(uint64(math.MaxUint64)))
+
+	// rand should be 0.813394603644003, a positive overflow happens
+	p.Generator = &distuv.Laplace{Mu: p.Mu, Scale: p.Scale, Src: rand.NewSource(2)}
+	require.Equal(t, math.MaxInt64, p.addNoise(math.MaxInt64))
+	require.Equal(t, uint64(math.MaxUint64), p.addNoise(uint64(math.MaxUint64)))
+}
+
+// Verifies that even addNoise() modifies 0 values as well
+func TestAddNoiseWithZeroValue(t *testing.T) {
+	p := Noise{
+		NoiseType: "laplace",
+		Scale:     1.0,
+		Log:       testutil.Logger{},
+	}
+	_ = p.Init()
+	noise := p.addNoise(0)
+	require.NotEqual(t, noise, 0)
```

Comment:
This makes a flaky test as you cannot _guarantee_ that the noise is not zero by chance.